### ISSUE TITLE
CKAN gets confused by dependency

### ISSUE THREAD

**Issue by SkyMarshal** [KSP-CKAN/CKAN#1189]
Selected mods to install:
![ksp1](https://cloud.githubusercontent.com/assets/1243563/8370060/cd764ee4-1b77-11e5-9436-2bc7fde9f362.png)

The mods are downloaded and preped to be installed... then this shows up:
![ksp2](https://cloud.githubusercontent.com/assets/1243563/8370067/ec119188-1b77-11e5-8efb-1786a692225b.png)

Which is freaking weird as that, you will notice, was already selected... but it gets weirder.  Reaffirming my desire to get the stock config, it appears that everything worked...
![ksp3](https://cloud.githubusercontent.com/assets/1243563/8370088/30c1129a-1b78-11e5-85b1-7aaf3dc18aff.png)

...except, what's that?  Go to changes?  But they're all applied!
![ksp4](https://cloud.githubusercontent.com/assets/1243563/8370099/49d114c4-1b78-11e5-9938-cd46fde5b01b.png)

Attempting to install that sets you back to the second image.  No idea what's happening.

**Comment by pjf** [comment]
Oh my. I'm guessing something odd is going on during relationship resolution. :/

We start off with a pretty big changeset here; if it's convenient for you, is it possible to see you can reproduce the behaviour with a smaller number of mods selected for install? That might give us an insight as to exactly where things are going weird.

**Comment by SkyMarshal** [comment]
What the fuck.  I figured it out.  It ONLY happened when FAR wanted to add the default ships... the issue was that I had nuked the %KSP%/Ships folder.  When I installed FAR separately, it gave me a "could not find part of path" error.  Upon resolving the issue (creating a directory structure identical to the normal KSP ships folder), FAR installed correctly.

HOWEVER.  To verify that that was the issue, I uninstalled FAR and installed FAR + Distant Object Enhancement.... same error.  So I tried installing FAR again.  Since the folders were empty after it uninstalled FAR... it hammered my recreated ships folder flat, causing the error to occur again when I tried to reinstall it.

TL;DR: Missing KSP folder structure confused FAR and masqueraded as a dependency error.  Fixing folder structure let stuff get installed correctly, but it turns out CKAN will remove important directories if they get emptied during uninstall.

(p.s. recreating the folder a second time let FAR and DOE install without issue)

**Comment by pjf** [comment]
@SkyMarshal : You are *awesome*.

>  Since the folders were empty after it uninstalled FAR... it hammered my recreated ships folder flat

Oh dang, nooo! CKAN, don't do *that*.

Opening a bug, because this is definitely one. (And which should be easy to fix)

**Comment by SkyMarshal** [comment]
I can honestly say I would never have guessed that to be the cause.  As the true culprit has been apprehended--and I got all the mods to install--I am closing this.
